### PR TITLE
Default settings for text-editors, who support `editorconfig.org`

### PR DESCRIPTION
**Pull request by maxkoryukov**
#### Database Migration
NO

#### Description
Default settings for text-editors, who support `editorconfig.org`

**Base styleguide**
* UTF-8
* `lf` - line endings (linux)
* 4 spaces for indentation
 
#### Todos
- [x] Tests (_not required_)
- [x] Documentation (Already documented in wiki)

#### Issues Fixed or Closed by this PR

* Nope

### CODE REVIEW

**Comment by maxkoryukov** [comment]
I have no idea, why the build failed on TeamCity (I don't have an access)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# editorconfig.org
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 4
```
Comment (markus101) [inline .editorconfig]: The current repo is a bit messy, but we've decided to move to 2 spaces for indentation.

**Comment by markus101** [comment]
You should have guest access, but the issue isn't with your changes something that we need to check on.

Thanks for adding this.

File: .editorconfig (hunk 1 above)
Comment (markus101) [inline .editorconfig]: Looks like we're currently using 4 everywhere except less, in a future version of the UI we'll be switching to 2 spaces, but we can make that change later.

**Comment by Taloth** [comment]
I use editorconfig myself. I've been meaning to add .editorconfig.

I strongly recommend you avoid the wildcard. just map *.cs, *.html, *.js and *.less as appropriate.
Atm, cs,html,js,hbs = 4 spaces. .less = 2 spaces.

Also, we use crlf, and commit lf. git takes care of that automatically on checkin/checkout.

**Comment by maxkoryukov** [comment]
@Taloth, @markus101, guys, I am new in Sonarr, and for this PR I have used only info from the [**wiki** page](https://github.com/Sonarr/Sonarr/wiki/Contributing)

I will be happy to improve this PR, so now, I am going to add another one commit, and edit wiki page;)

File: .editorconfig (hunk 1 above)
Comment (maxkoryukov) [inline .editorconfig]: tuned!

**Comment by maxkoryukov** [comment]
@Taloth , I just downloaded one raw `.cs` file from GitHub (don't know other way to get a file with untouched lineendings from github), and checked the LF vs. CRLF. And there is just LF symbol.

So, I leave:
```ini
end_of_line = lf
```
in the `.editorconfig`

**Comment by maxkoryukov** [comment]
* [x] Wiki page done

**Comment by Taloth** [comment]
Git automatically converts crlf to lf when checking in. So the repository is lf but the editor is crlf.

**Comment by markus101** [comment]
> Git automatically converts crlf to lf when checking in. So the repository is lf but the editor is crlf.

For the UI side of things it's not too bad to keep it LF, it makes linters happier and CRLF -> LF on check-in is primarily for Windows.

**Comment by maxkoryukov** [comment]
CR

**Comment by maxkoryukov** [comment]
LF

**Comment by maxkoryukov** [comment]
Just a voting;)

**Comment by Taloth** [comment]
Let git autocrlf deal with it. Changing it at this point is like inviting disaster. Everybody would have to change their git settings, check everything out again etc.
Adding a .editorconfig is fine by me, but don't change the rules.

I won't discuss it any further.

**Comment by Taloth** [comment]
Closed via https://github.com/Sonarr/Sonarr/commit/206ff12b7159d922f4b9820aa7cb47ab0e524e35

**Comment by maxkoryukov** [comment]
but why? there is the same settings???

I don't understand...

**Comment by Taloth** [comment]
Without the cr stuff. As I said before, I wanted it to stick to the original coding rules.

**Pull request closed without merging**